cfe_internal/core/watchdog/templates/watchdog.mustache

```diff
@@ -1,8 +1,320 @@
+    echo "$(date) Found cf-execd not running" >> ${LOGFILE}
+    echo "- cf-execd not running" >> ${COLLECTION_REPORT}
+    PATHOLOGY_COUNT=$((${PATHOLOGY_COUNT}+1))
+fi
+
+# Pathology #1.5: More than one cf-execd is running.
+if [ "${_COUNT_CF_EXECD_PROCS}" -gt "1" ]; then
+    echo "$(date) Found ${_COUNT_CF_EXECD_PROCS} cf-execd processes running" >> ${LOGFILE}
+    echo "- Found ${_COUNT_CF_EXECD_PROCS} cf-execd running" >> ${COLLECTION_REPORT}
+    PATHOLOGY_COUNT=$((${PATHOLOGY_COUNT}+1))
+fi
+
+# Pathology #2: cf-agent execution times exceed body executor control agent_expireafter
+# cf-agent processes launched by cf-execd have not communicated back to cf-execd
+# within $(agent_expireafter). This may be OK, but may also indicate that
+# cf-agent is hung.
+# Outputs could be empty because:
+# - cf-execd hasn't executed cf-agent since purging outputs
+# - cf-agent is not producing output (the expected normal state)
+if [ "$(ls -A ${OUTPUTS_DIR})" ]; then
+    observation=$(find "${OUTPUTS_DIR}" ! -name previous | xargs grep "cf-execd: timeout waiting for output from agent")
+    if [ ${#observation} -gt 0 ]; then
+        count=$(expr 0 + $(echo -n "${observation}" | wc -l))
+        echo "$(date) Found ${count} occurrences of cf-execd terminating unresponsive cf-agent" >> ${LOGFILE}
+        echo "- ${count} cf-agent terminations" >> ${COLLECTION_REPORT}
+        PATHOLOGY_COUNT=$((${PATHOLOGY_COUNT}+1))
+    fi
+fi
+
+# Pathology #3: cf-agent process(s) running longer than expected
+# While not strictly a pathology, this may indicate a hung agent
+# On AIX, etime (and etimes) is displayed in human readable form
+# e.g:
+#    USER      PID     ELAPSED COMMAND
+#    root 10551366  5-00:35:58 /var/cfengine/bin/cf-agent
+# We have to convert that into seconds so that we can determine if it's been
+# running longer than we expect, that's what the second awk command is for.
+observation=$(ps -e -o user,pid,etime,args | awk 'FNR == 1 {next} /\/var\/cfengine\/bin\/cf-agent/ {print $3}' | awk 'BEGIN { FS = ":" } \
+{
+  if (NF == 2) {
+    etimes = $1*60 + $2
+  } else if (NF == 3) {
+    split($1, a, "-");
+    if (a[2] != "" ) {
+      etimes = ((a[1]*24+a[2])*60 + $2) * 60 + $3;
+    } else {
+      etimes = ($1*60 + $2) * 60 + $3;
+    }
+  }
+ if (etimes > 300) {
+   print
+}
+}')
+if [ ${#observation} -gt 0 ]; then
+    count=$(expr 1 + $(echo -n "${observation}" | wc -l))
+    echo "$(date) Found ${count} cf-agent processes running longer than 300s" >> ${LOGFILE}
+    echo "- ${count} cf-agent processes running longer than 300s" >> ${COLLECTION_REPORT}
+    PATHOLOGY_COUNT=$((${PATHOLOGY_COUNT}+1))
+fi
+
+# Pathology #4: High number of concurrent cf-agent processes
+# While cf-agent is designed to be able to run concurrently, having many
+# concurrent cf-agent processes may indicate an an issue like an agent pile up
+observation=$(ps -e -o pid,etime,args | awk '/\/var\/cfengine\/bin\/cf-agent/ {print}')
+if [ ${#observation} -gt 0 ]; then
```

Comment:
Same here with portability. Please double-check this is portable and replace it with `wc -l` if not.